### DISCUSSION TITLE
How to resolve incompatibility with 19.2 upgrade

### DISCUSSION THREAD

**Discussion by maanmusleh** [microsoft/ALAppExtensions#15271] [Q&A]
Hello All, I need help!

We have received an email from MS stating that our extension needs updating.
That's the error message
src/tableext/WarehouseRequest.TableExt.al (5,22) - Error AL0155: A member of type Field with name 'Shipping Agent Service Code' is already defined in Table 'Warehouse Request' by the extension 'Base Application by Microsoft (19.2.32678.0)'.

Quick Background:
In a previous version we have extended warehouse request table to add field "Shipping Agent Service code".
The problem is that Base business central 19.2 has added that field to the same table and that;s creating the schema conflict

I am unable to rename or remove the fielld , so how do I resolve this incompatibility issue?

**Comment by SimonOfHH** [comment]
Hi @miracleitconsulting 

First of all, I'm not sure if this is the right place for this kind of discussion. I guess the [Yammer-group](https://www.yammer.com/dynamicsnavdev/), the "[#BCALHelp](https://twitter.com/search?q=%23bcalhelp)"-tag on Twitter or the Discord-channel "[BC Professionals](https://discord.gg/4wbfNv3)" might be better places for this (because the ALAppExtensions-repo is not for the Base-App). But I might be wrong.

Anyway, regarding your question: I think the way to go here, is to mark your custom field as "ObsoleteState = Pending" in a first step. You could/should also add an upgrade-routine to transfer data from your custom field to the new standard-field. With another version-update of your app, you can then set the "ObsoleteState = Removed".

And I guess next time you should remember to prefix/suffix all your custom fields to avoid these kind of problems 😉

Hope this helps!

**Comment by miljance** [comment]
That is a think a PTE? Otherwise it would not reach the App Source without the affix. Therefore I would in a sandbox try the following: Deinstall and unpublish the PTE, upgrade the sandbox and then install the PTE without the field. Should work. At least I hope.

**Comment by miljance** [comment]
I think the the previous Idea will not work. It has fundamental flaws.
Second try: 
- Move the field value to the new temporary extension.
- Upload the new PTE without the field using Force Option in order to remove the field value from the PTE.
![image](https://user-images.githubusercontent.com/42636293/142841793-65a02d12-eb8f-4065-86f4-2a847b421d05.png)

- Run the tenant upgrade
- Restore the value from the temporary extension to the new field standard field in the Base App.

**Comment by maanmusleh** [comment]
Thanks @miljance and @SimonOfHH for your replies.

@miljance i'm really hoping this will work. I do not need to copy data from old field as Web Request table is in itself kind of a staging table between sales and warehouse, so this way I can avoid creating a temporary extension.

I will try this approach for the coming 19.0 to 19.2 upgrade and report back here if it worked.

**Comment by miljance** [comment]
You are welcome. As far as I can tell, the great guy from Microsoft Support named Duilio have the similar approach:
![image](https://user-images.githubusercontent.com/42636293/142874701-f35360ab-ea99-44f8-b364-c608f1527dfb.png)
